okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.java

```diff
@@ -216,12 +218,24 @@ public static Platform buildIfSupported() {
       OptionalMethod<Socket> getAlpnSelectedProtocol = null;
       OptionalMethod<Socket> setAlpnProtocols = null;
 
-      // Attempt to find Android 5.0+ APIs.
-      try {
-        Class.forName("android.net.Network"); // Arbitrary class added in Android 5.0.
-        getAlpnSelectedProtocol = new OptionalMethod<>(byte[].class, "getAlpnSelectedProtocol");
-        setAlpnProtocols = new OptionalMethod<>(null, "setAlpnProtocols", byte[].class);
-      } catch (ClassNotFoundException ignored) {
+      final OptionalMethod<Socket> getAlpnMethod
```

Comment:
I like this. How about the method name `supportsALPN()` instead of `isNewerAndroid()` since that's what we're checking for.

Happy to make this change. Thanks @yschimke.